src/main/java/com/google/devtools/build/lib/sandbox/AbstractSandboxSpawnRunner.java

```diff
@@ -201,7 +204,7 @@ private final SpawnResult run(
     FileOutErr outErr = context.getFileOutErr();
     Duration timeout = context.getTimeout();
 
-    SubprocessBuilder subprocessBuilder = new SubprocessBuilder();
+    SubprocessBuilder subprocessBuilder = new SubprocessBuilder(clientEnv);
```

Comment:
What's the advantage of passing `clientEnv` to the constructor when it later gets overridden by `setEnv`? Would it be clearer to have only `setEnv` and make an explicit choice at every callsite?